### PR TITLE
Add image-remove module

### PR DESCRIPTION
**Pull request by ovekyc**
* Add testcode
* Edit mock-data

In Progress,
Item remove logic will be added.

### CODE REVIEW

File: src/server/images/controllers.js [hunk 1]
```diff
@@ -134,5 +134,100 @@ export default {
...
       return cb(new APIError(err, {statusCode: err.statusCode, message: err.message}));
     });
+  },
+  remove(req, res, cb) {
+    const {userKey} = req.headers;
+    const imageKey = req.params.id;
+    assert(userKey, 'user-key should be provided');
+    assert(imageKey, 'image-key should be provided');
+
+    let itemKey;
```
Comment (festinalente91) [inline src/server/images/controllers.js:151]: now I can get why we need async-await :)

File: src/server/images/controllers.js [hunk 2]
```diff
@@ -134,5 +134,123 @@ export default {
...
+        });
+      });
+    }).then(() => {
+      // count image of item
+      const prefixes = [];
+      const promises = [];
+      prefixes.push(KeyUtils.getPrefix(ENTITY.IMAGE, STATE.ALIVE, itemKey));
+      prefixes.push(KeyUtils.getPrefix(ENTITY.IMAGE, STATE.EXPIRED, itemKey));
+
+      prefixes.map((prefix) => {
```
Comment (festinalente91) [inline src/server/images/controllers.js]: you can just return Promise.all(prefixes.map((prefix) => { return new Promise() }));

Comment (ovekyc) [reply src/server/images/controllers.js]: #157

**Comment by hewonjeong** [comment]
[As a REST client, I'd like to remove my image](https://trello.com/c/wbuNPFkt/193-as-a-rest-client-i-d-like-to-remove-my-image)

File: src/server/images/controllers.js [hunk 3]
```diff
@@ -134,5 +134,115 @@ export default {
...
+        { type: 'del', key: createdPostKey }
+      ];
+      return new Promise((resolve, reject) => {
+        db.batch(opts, (err) => {
+          return err ? reject(err) : resolve();
+        });
+      });
+    }).then(() => {
+      // add REMOVED state
+      function replaceAt(str, index, char) {
```
Comment (festinalente91) [inline src/server/images/controllers.js]: please define this function outside. As you know defining a function in a function is not that good. :)

**Review by festinalente91 — APPROVED** [review]

**Pull request merged**